### PR TITLE
Bar series now correctly applied the y0 value

### LINKED ISSUE #397: bar series incorrectly computes y0 value
if you specify a non-zero y0 value the whole bar series is offset rather than just the baseline.

### PR DESCRIPTION
**Pull request by ColinEberhardt**
Previously y0 was been included in teh bar top and bar bottom calculation, as a result, the two y0's were cancelled out when computng bar height. Doh!

Fixes #397

### CODE REVIEW

File: src/series/bar.js [hunk 1]
```diff
@@ -27,9 +27,8 @@
                     .height(0);
 
                 var x = function(d, i) { return xValueScaled(d, i); },
-                    y0 = function(d, i) { return y0Value(d, i); },
-                    barTop = function(d, i) { return yScale(y0(d, i) + y1Value(d, i)); },
-                    barBottom = function(d, i) { return yScale(y0(d, i)); };
+                    barTop = function(d, i) { return yScale(y1Value(d, i)); },
```
Comment (chrisprice) [inline src/series/bar.js]: Minor point but should barTop and barBottom now be renamed y0 and y1?

File: visual-tests/src/test-fixtures/series/bar.js [hunk 2]
```diff
@@ -25,18 +25,32 @@
...
-        .yScale(priceScale);
+        .yScale(priceScale1);
 
     // Add it to the chart
     container.append('g')
         .datum(data)
-        .call(bar);
+        .call(bar1);
+
+    // Create scale for y axis
```
Comment (chrisprice) [inline visual-tests/src/test-fixtures/series/bar.js]: You might have avoided this intentionally but I think you could just reuse the price scale with a new range?

File: src/series/bar.js (hunk 1 above)
Comment (ColinEberhardt) [inline src/series/bar.js]: Good point - the naming is probably a left-over from when the bar series didn't support negative values.

File: visual-tests/src/test-fixtures/series/bar.js (hunk 2 above)
Comment (ColinEberhardt) [inline visual-tests/src/test-fixtures/series/bar.js]: I don't know ... mutating and re-using a scale feels wrong. I know it works, but just doesn't feel right!

Comment (jleft) [inline visual-tests/src/test-fixtures/series/bar.js]: If you haven't already seen, D3's scales have [`.copy()` ](https://github.com/mbostock/d3/wiki/Quantitative-Scales#linear_copy), which:

> Returns an exact copy of this linear scale. Changes to this scale will not affect the returned scale, and vice versa.

So you could do something like this (which looks quite neat to me):

```js
var priceScale2 = priceScale1.copy()
    .range([height / 2, height]);
```

Comment (ColinEberhardt) [inline visual-tests/src/test-fixtures/series/bar.js]: > If you haven't already seen, D3's scales have .copy()

I'd forgotten about that, thanks for the reminder. Interesting that D3 scales have a copy function, whereas most other components do not.

**Pull request merged**